src/transformers/models/swiftformer/modeling_tf_swiftformer.py

```diff
@@ -0,0 +1,911 @@
+            training=training,
+        )
+        return outputs
+
+    def build(self, input_shape=None):
+        if self.built:
+            return
+        self.built = True
+        if getattr(self, "swiftformer", None) is not None:
+            with tf.name_scope(self.swiftformer.name):
+                self.swiftformer.build(None)
+
+
+@add_start_docstrings(
+    """
+    TFSwiftFormer Model transformer with an image classification head on top (e.g. for ImageNet).
+    """,
+    TFSWIFTFORMER_START_DOCSTRING,
+)
+class TFSwiftFormerForImageClassification(TFSwiftFormerPreTrainedModel):
+    def __init__(self, config: SwiftFormerConfig, **kwargs) -> None:
+        super().__init__(config, **kwargs)
+
+        self.num_labels = config.num_labels
+        self.swiftformer = TFSwiftFormerMainLayer(config, name="swiftformer")
+
+        # Classifier head
+        self.norm = tf.keras.layers.BatchNormalization(
+            epsilon=config.batch_norm_eps, momentum=0.9, name="norm"
+        )  # FIXME
+        self.head = (
+            tf.keras.layers.Dense(self.num_labels, name="head")
+            if self.num_labels > 0
+            else tf.keras.layers.Identity(name="head")
+        )
+        self.dist_head = (
+            tf.keras.layers.Dense(self.num_labels, name="dist_head")
+            if self.num_labels > 0
+            else tf.keras.layers.Identity(name="dist_head")
+        )
+
+    @unpack_inputs
+    @add_start_docstrings_to_model_forward(TFSWIFTFORMER_INPUTS_DOCSTRING)
+    def call(
+        self,
+        pixel_values: Optional[tf.Tensor] = None,
+        labels: Optional[tf.Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        training: bool = False,
+    ) -> Union[tuple, TFImageClassifierOutputWithNoAttention]:
+        r"""
+        labels (`tf.Tensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the image classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        # run base model
+        outputs = self.swiftformer(
+            pixel_values,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+        )
+
+        sequence_output = outputs.last_hidden_state if return_dict else outputs[0]
+        sequence_output = tf.transpose(sequence_output, perm=[0, 2, 3, 1])
+
+        # run classification head
+        sequence_output = self.norm(sequence_output, training=training)
+        # FIXME: review this conversion
+        sequence_output = tf.transpose(sequence_output, perm=[0, 3, 1, 2])
+        _, c, h, w = sequence_output.shape
```

Comment:
No single letter var names - they should all be clear and explicit 

```suggestion
        _, num_channels, height, width = sequence_output.shape
```

Reply:
Addressed in https://github.com/huggingface/transformers/pull/23342/commits/3d6f3dd1aa4cfa2ba3a48c1dfe7e1355a0b4a8bb